File: core/src/main/java/com/ustadmobile/core/impl/http/UmHttpException.java
package com.ustadmobile.core.impl.http;

import java.io.IOException;

/**
 * Created by mike on 12/26/17.
 */

public class UmHttpException extends IOException {

    private UmHttpResponse response;

    private Exception rootCause;

    public UmHttpException(UmHttpResponse response) {
        this.response = response;
    }

    public UmHttpException(Exception rootCause) {
        this.rootCause = rootCause;
    }

    public int getStatus() {
        if(response != null)
            return response.getStatus();
        else
            return -1;
    }

    public Exception getRootCause() {
        return rootCause;
    }

}

File: core/src/main/java/com/ustadmobile/core/impl/http/UmHttpException.kt
package com.ustadmobile.core.impl.http

import java.io.IOException

/**
 * Created by mike on 12/26/17.
 */

class UmHttpException : IOException {

    private var response: UmHttpResponse? = null

    var rootCause: Exception? = null

    val status: Int
        get() = response?.status ?: -1

    constructor(response: UmHttpResponse) {
        this.response = response
    }

    constructor(rootCause: Exception) {
        this.rootCause = rootCause
    }

}
